### PR TITLE
Fix issues in the user attribute section

### PR DESCRIPTION
**Pull request by DinikaSen**
### Purpose
This PR is to fix the issues mentioned in https://github.com/wso2/product-is/issues/11912 in the user attribute section of the Console's Application section

### Related Issues
- https://github.com/wso2/product-is/issues/11912

### Checklist
- [x] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on final implementation
- [ ] Documentation provided. (Add links)
- [ ] Unit tests provided. (Add links if any)
- [ ] Integration tests provided. (Add links if any)

### Related PRs
- Related PR #1

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx [hunk 1]
```diff
@@ -719,14 +744,13 @@ export const AttributeSelection: FunctionComponent<AttributeSelectionPropsInterf
                                                                             claimMappingOn={ claimMappingOn }
                                                                             claimMappingError={ claimMappingError }
                                                                             readOnly={
-                                                                                (selectedSubjectValue === claim.claimURI)
+                                                                                (selectedSubjectValue ===
```
Comment (NipuniBhagya) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx:747]: Let's format this line properly

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx [hunk 2]
```diff
@@ -719,14 +744,13 @@ export const AttributeSelection: FunctionComponent<AttributeSelectionPropsInterf
...
+                                                                                    resolveClaimValue(claim.claimURI))
                                                                                     ? true
                                                                                     : readOnly
                                                                             }
-                                                                            subject={ claimMappingOn
-                                                                                ? selectedSubjectValue === getCurrentMapping(
-                                                                                    claim?.claimURI)?.applicationClaim
-                                                                                : selectedSubjectValue === claim.claimURI
+                                                                            subject={ selectedSubjectValue ===
```
Comment (NipuniBhagya) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx:752]: Let's properly format this line too.

**Review by NipuniBhagya — APPROVED** [review]

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ import isEmpty from "lodash-es/isEmpty";
 import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
 import { useTranslation } from "react-i18next";
 import { useDispatch } from "react-redux";
-import { Checkbox, Icon, Input, Label, Popup, Table } from "semantic-ui-react";
+import {Checkbox, Divider, Icon, Input, Label, Popup, Table} from "semantic-ui-react";
```
Comment (inthirakumaaran) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx:25]:
```suggestion
import { Checkbox, Divider, Icon, Input, Label, Popup, Table } from "semantic-ui-react";
```

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx [hunk 4]
```diff
@@ -308,16 +315,26 @@ export const AttributeSelection: FunctionComponent<AttributeSelectionPropsInterf
...
+                        return;
+                    }
+                    if (claim.localClaim.uri === defaultSubjectAttribute &&
+                        isSubjectClaimSetToDefaultWithoutMapping()) {
+                        return;
+                    }
                     const claimMapping: ExtendedClaimMappingInterface = {
                         addMapping: true,
-                        applicationClaim: claim?.applicationClaim,
+                        applicationClaim: claim.applicationClaim,
```
Comment (inthirakumaaran) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx:327]: Do we need to remove these undefined checks?

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 5]
```diff
@@ -317,15 +318,51 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
...
+                const newClaimMapping: ExtendedClaimMappingInterface = {
+                    addMapping: false,
+                    applicationClaim: "",
+                    localClaim: {
+                        displayName: claim.displayName,
+                        id: claim.id,
+                        uri: claim.claimURI
+                    }
+                };
+                if (!(claimMappingList.some((claimMap) => claimMap.localClaim.uri === claim.claimURI))) {
```
Comment (inthirakumaaran) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:334]:
```suggestion
                if (!(claimMappingList.some((claimMap: ExtendedClaimMappingInterface) => claimMap.localClaim.uri === claim.claimURI))) {
```

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 6]
```diff
@@ -317,15 +318,51 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
...
+                    claimMappingList.push(newClaimMapping);
+                }
+            });
+
+            removedClaims.map((claim) => {
```
Comment (inthirakumaaran) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:339]:
```suggestion
            removedClaims.map((claim: Claim) => {
```

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 7]
```diff
@@ -392,27 +429,51 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
...
-                            key: element?.localClaim?.id,
+                    if (!element || !element.localClaim) {
+                        return;
+                    }
+                    const option: DropdownOptionsInterface = {
+                        key: element?.localClaim.uri,
+                        text: (
+                            <SubjectAttributeListItem
+                                key={ element.localClaim.id }
+                                displayName={ element.applicationClaim ?
```
Comment (inthirakumaaran) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:444]: Add undefined check

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx (hunk 4 above)
Comment (DinikaSen) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx:327]: With the below section added(line 318), we no longer need to have those undefuned checks.
```
if (!claim || !claim.localClaim) {
         return;
}
```

**Review by inthirakumaaran — APPROVED** [review]

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx (hunk 7 above)
Comment (DinikaSen) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:444]: Undefined check is added in the beginning of the map function

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx (hunk 1 above)
Comment (DinikaSen) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx]: Addressed with https://github.com/wso2/identity-apps/pull/2136/commits/ee4252ee9ebc4c53695c6349e0daed00b03ebf15

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx (hunk 2 above)
Comment (DinikaSen) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx]: Addressed with https://github.com/wso2/identity-apps/pull/2136/commits/ee4252ee9ebc4c53695c6349e0daed00b03ebf15

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx (hunk 6 above)
Comment (DinikaSen) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx]: Addressed with https://github.com/wso2/identity-apps/pull/2136/commits/284eb33d3d3c2daa07cc36512ccf7b83cad4eac1

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx (hunk 5 above)
Comment (DinikaSen) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx]: Addressed with https://github.com/wso2/identity-apps/pull/2136/commits/284eb33d3d3c2daa07cc36512ccf7b83cad4eac1

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 8]
```diff
@@ -317,15 +318,51 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
         }
     };
 
-    const removeMapping = (claim: Claim) => {
+    const updateMappings = (addedClaims: Claim[], removedClaims: Claim[]) => {
+        if (selectedDialect.localDialect) {
+            const claimMappingList: ExtendedClaimMappingInterface[] = [...claimMapping];
+            addedClaims.map((claim) => {
```
[the remark this answers is not in the document]
Comment (DinikaSen) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx]: Addressed with https://github.com/wso2/identity-apps/pull/2136/commits/284eb33d3d3c2daa07cc36512ccf7b83cad4eac1

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx (hunk 3 above)
Comment (DinikaSen) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx]: Addressed with https://github.com/wso2/identity-apps/pull/2136/commits/284eb33d3d3c2daa07cc36512ccf7b83cad4eac1

**Pull request merged**